### ISSUE TITLE
Loop compiling

### ISSUE THREAD

**Issue by benjaminbours** [webpack-contrib/copy-webpack-plugin#580]
- Operating System: Ubuntu 20.04
- Node Version: 12.19
- NPM Version: 6.14.8
- webpack Version: 5.19
- copy-webpack-plugin Version: 7.0

### Expected Behavior

I would like compilation to happen only one time when I do only one change.

### Actual Behavior

The compilation happen multiples time, it's inconsistent, sometimes only 2 or 3 times, sometimes 8.
![2021-02-02_15-41](https://user-images.githubusercontent.com/8804297/106619576-05ec1580-6571-11eb-8f39-eb6bb0e4c474.png)

### Code
Here is the relevant part of my webpack.config.js
```js
  new CopyWebpackPlugin({
    patterns: [
      {
        from: path.join(__dirname, 'sources/media'),
        to: path.join(outputPath, 'media'),
        toType: 'dir',
        transform: (content, path) => optimizeAsset(content, path),
      },
      {
        from: path.join(__dirname, 'libraries/project/media'),
        to: path.join(outputPath, 'media'),
        toType: 'dir',
        transform: (content, path) => optimizeAsset(content, path),
      },
      {
        from: path.join(__dirname, 'libraries/vendors/media'),
        to: path.join(outputPath, 'media'),
        toType: 'dir',
        transform: (content, path) => optimizeAsset(content, path),
      },
      // Shared
      {
        from: path.join(
          __dirname,
          `node_modules/react-dom/umd/react-dom.${isDevelopment ? 'development' : 'production.min'}.js`
        ),
        to: path.join(outputPath, 'media/js'),
        toType: 'dir',
      },
      {
        from: path.join(
          __dirname,
          `node_modules/react/umd/react.${isDevelopment ? 'development' : 'production.min'}.js`
        ),
        to: path.join(outputPath, 'media/js'),
        toType: 'dir',
      },
      {
        from: path.join(
          __dirname,
          `node_modules/react-router-dom/umd/react-router-dom${isDevelopment ? '' : '.min'}.js`
        ),
        to: path.join(outputPath, 'media/js'),
        toType: 'dir',
      },
      {
        from: path.join(__dirname, 'private_path/nexus-client/client.js'),
        to: path.join(outputPath, 'media/js/nexus-client.js'),
        toType: 'file',
      },
    ],
    options: { concurrency: 200 },
  })
```

### How Do We Reproduce?

I just do `webpack serve --hot --mode development` to start my development server. When I'm editing a target of the copy-plugin, it creates compiling loop. I used this code the log what trigger a compilation.

```js
class WatchRunPlugin {
  apply(compiler) {
    compiler.hooks.watchRun.tap('WatchRun', (comp) => {
      if (comp.modifiedFiles) {
        const changedFiles = Array.from(comp.modifiedFiles, (file) => `\n  ${file}`).join('');
        console.log('===============================');
        console.log('FILES CHANGED:', changedFiles);
        console.log('===============================');
      }
    });
  }
};
```
As you can see, in the screenshot above, it detect the first file change, then multiple compilation are executed based on no file changed apparently. I would like to avoid this, as the loop are sometimes making crash the web page in my browser. Is that related to this plugin? Is that related to webpack? When editing file related to an entry point and not to the copy plugin, it compiles only once, and it seems to be the expected behavior, that's why I suspect this plugin to be the cause of this.

I recently update to webpack 5 and that's when trouble started to happen. No issue with webpack 4.

**Comment by alexander-akait** [comment]
Because you change files inside `from`, no magic here, sorry you don't provide enough information to example which one from these patterns was changed, you can found a lot of `loop` issues (just use search) where I provide on problems, without reproducible I can't help here, sorry, won't fix

**Comment by alexander-akait** [comment]
If you provide more information I will help

**Comment by benjaminbours** [comment]
Well from the screenshot I provided we could see this is this pattern.
```js
      {
        from: path.join(__dirname, 'sources/media'),
        to: path.join(outputPath, 'media'),
        toType: 'dir',
        transform: (content, path) => optimizeAsset(content, path),
      },
```

> Because you change files inside from, no magic here

Do you mean this is the expected behavior when you edit something 'watched' by the copy plugin?

**Comment by alexander-akait** [comment]
> Do you mean this is the expected behavior when you edit something 'watched' by the copy plugin?

You emit something in this directory and do change after emitting so webpack run again, from my experience I can say that if you want to solve a problem quickly, then I recommend to create reproducible test repo, it is very hard to say, I even don't know your full plugin list
